src/profile/helpers/__tests__/updateMutationVariables.test.ts

```diff
@@ -106,112 +108,251 @@ const newAddress = {
-  language: Language.FINNISH,
-  primaryEmail,
-  emails: [primaryEmail, newEmail, updatedSecondaryEmail],
-  phones: [primaryPhone, newPhone, updatedSecondaryPhone],
-  primaryPhone,
-  addresses: [primaryAddress, newAddress, updatedSecondaryAddress],
-  primaryAddress,
-};
+type FormParts = (keyof FormValues)[];
 
-test('add arrays are formed correctly with new data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
+const multiNodeFormParts: FormParts = [
+  'primaryAddress',
+  'primaryEmail',
+  'primaryPhone',
+  'phones',
+  'emails',
+  'addresses',
+];
 
-  expect(variables.input.profile.addAddresses).toEqual([
-    getAddressAsComparisonObject(newAddress),
-  ]);
+const nameFormParts: FormParts = ['firstName', 'nickname', 'lastName'];
 
-  expect(variables.input.profile.addEmails).toEqual([
-    getEmailAsComparisonObject(newEmail),
+const getFormValues = (parts: FormParts): Partial<FormValues> => {
+  const myProfileNamesAndLanguage = _.pick(myProfile.myProfile, [
+    ...nameFormParts,
+    'language',
   ]);
+  const formValues = {
+    ...myProfileNamesAndLanguage,
+    primaryEmail,
+    emails: [primaryEmail, newEmail, updatedSecondaryEmail],
+    phones: [primaryPhone, newPhone, updatedSecondaryPhone],
+    primaryPhone,
+    addresses: [primaryAddress, newAddress, updatedSecondaryAddress],
+    primaryAddress,
+  };
+  return _.pick(formValues, parts) as Partial<FormValues>;
+};
 
-  expect(variables.input.profile.addPhones).toEqual([
-    getPhoneAsComparisonObject(newPhone),
-  ]);
-});
+describe('Update variables for basic data (names) are formed correctly', () => {
+  let formValues: Partial<FormValues>;
 
-test('add arrays are empty when using existing data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    {
-      ...formValues,
-      addresses: [primaryAddress, updatedSecondaryAddress],
-      emails: [primaryEmail, updatedSecondaryEmail],
-      phones: [primaryPhone, updatedSecondaryPhone],
-    },
-    myProfile
-  );
-
-  expect(variables.input.profile.addPhones).toEqual([]);
-  expect(variables.input.profile.addEmails).toEqual([]);
-  expect(variables.input.profile.addAddresses).toEqual([]);
-});
+  beforeEach(() => {
+    formValues = getFormValues(nameFormParts);
+  });
 
-test('update arrays are formed correctly with updated data objects', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
-  expect(variables.input.profile.updateAddresses).toEqual([
-    getAddressAsComparisonObject(updatedSecondaryAddress),
-  ]);
-  expect(variables.input.profile.updateEmails).toEqual([
-    getEmailAsComparisonObject(updatedSecondaryEmail),
-  ]);
-  expect(variables.input.profile.updatePhones).toEqual([
-    getPhoneAsComparisonObject(updatedSecondaryPhone),
-  ]);
+  test('Variables in update data match formValues even if not changed', () => {
```

Comment:
There seems to be 2 different conventions for this test function: for example editData.test.ts applies function _it_. I wonder if it would be a good idea to only use one of them?

Reply:
Yes, "test"-funcs are used in the old project files. I kept the same func calls inside each file. I'll rename them.

Reply:
Done.